source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -113,6 +117,27 @@ ExtProcLoggingInfo::grpcCalls(envoy::config::core::v3::TrafficDirection traffic_
              : encoding_processor_grpc_calls_;
 }
 
+absl::flat_hash_map<std::string, Extensions::Filters::Common::Expr::ExpressionPtr>
+FilterConfig::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) const {
+  absl::flat_hash_map<std::string, Extensions::Filters::Common::Expr::ExpressionPtr> expressions;
+#if defined(USE_CEL_PARSER)
+  for (const auto& matcher : matchers) {
+    auto parse_status = google::api::expr::parser::Parse(matcher);
```

Comment:
The CEL usage here is not correct.

We need to extend the lifetime of`parse_status` here because [CEL lib specifically requires](https://github.com/google/cel-cpp/blob/e4067116de7b37efb62042fb4cd320a1877a2e51/eval/public/cel_expression.h#L89-L90) that parsed compiled expression(return of parser::Parse ) need to outlive the compiled expression (return of`Expr::createExpression`)

You can see some correct examples [here](https://github.com/envoyproxy/envoy/blob/2de016d1007aabff202220b8177167c9ab3e8c6a/source/extensions/common/wasm/foreign.cc#L161-L170) and [here](https://github.com/envoyproxy/envoy/blob/2de016d1007aabff202220b8177167c9ab3e8c6a/source/extensions/matching/input_matchers/cel_matcher/matcher.cc#L14-L22)